climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+                             ntime=None, detrend_by_control_unitialized=True,
+                             nmember_control=10):
+    """
+    Compute relative entropy.
+
+    Calculates EOFs from anomalies. Projects fields on EOFs to receive
+    pseudo-Principle Components per initialization and lead year. Calculate
+    relative entropy based on _relative_entropy_formula.
+
+    Args:
+        initialized (xr.Dataset): anomaly ensemble data with dimensions
+                                    initialization, member, time and spatial
+                                    [lon (x), lat(y)]. DPLE or PM_ds
+        control (xr.Dataset): anomaly control distribution with
+                                              non-spatial dimensions:
+                                              spatial [lon (x), lat(y)].
+                                              - LENS: member, time
+                                              - PM_control: initialization
+        anomaly_data (bool): Input data is anomaly alread. Default: False.
+        neofs (int): number of EOFs to use. Default: initialized.member.size.
+        curv (bool): if curvilinear grids are provided disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        detrend_by_control_uninitialized (bool): Default: True
+        nmember_control (int): number of members created from bootstrapping from control
+
+    Returns:
+        rel_ent (pd.DataFrame): relative entropy
+
+    """
+    # Defaults
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    # case if you only submit control with dim time, PM case
+    if ('ensemble' not in control.dims) and ('member' not in control.dims):
+        control_uninitialized = xr.concat([_bootstrap_dim(control, initialized.time.size, dim='member', dim_label=np.arange(
+            nmember_control)) for _ in range(initialized.initialization.size)], dim='initialization')
+        if isinstance(control_uninitialized, xr.DataArray):
+            control_uninitialized['initialization'] = initialized.initialization.values
+        elif isinstance(control_uninitialized, xr.Dataset):
+            control_uninitialized = control_uninitialized.assign(
+                initialization=initialized.initialization.values)
+
+    # case if you submit control with dim time and member, LENS case
+    elif 'member' in control.dims:
+        control_uninitialized = _bootstrap_dim(
+            control, initialized.time.size, time_dim='time', dim='initialization', dim_label=list(initialized.initialization.values))
+        #print('added member')
+
+    # initialized and control_uninitialized are allowed to have different dims
+    # as I need more members to sample my control distribution properly
+    # ToDo: understand this more
+    if initialized.dims != control_uninitialized.dims:
+        print(initialized.dims)
```

Comment:
Are these just for debugging? If not, do:
```python
print(list(initialized.dims))
```

since these are FrozenDictionaries they print out weird; example in the notebook.

Reply:
debugging